### DISCUSSION TITLE
Rust functions and methods hierarchy in Document Symbols

### DISCUSSION THREAD

**Discussion by telbizov** [folke/trouble.nvim#510] [General]
Hello,

Thanks for the great plugin and the rewrite. I started using the Document Symbols after the upgrade a couple of days ago and noticed something that I wanted to share. I suspect it might be a bug, but wanted to check here first before I file it as such.

In a Rust program if I have a struct with associated functions as well as methods, they don't show up under the structure's name and instead under the root of the tree in the symbols mode. Here's an example:

```rust
struct S1 {
    x: u64,
}
impl S1 {
    const fn new(x: u64) -> Self {
        Self { x }
    }
    fn method(&self) {
        println!("method");
    }
}
```

It is shown as:
```
 Document Symbols  11
  └╴  src/main.rs  11
    ├╴󰆼  S1 struct S1 { [8, 8]
    │ └╴  x x: u64, [9, 5]
    ├╴󰊕  new const fn new(x: u64) -> Self { [12, 14]
    ├╴󰊕  method fn method(&self) { [15, 8]
```

I would have expected, both new and method to be shown under the S1 tree sort of like this:
```
 Document Symbols  11
  └╴  src/main.rs  11
    ├╴󰆼  S1 struct S1 { [8, 8]
    │ └╴  x x: u64, [9, 5]
      ├╴󰊕  new const fn new(x: u64) -> Self { [12, 14]
      ├╴󰊕  method fn method(&self) { [15, 8]
```

Is it expected to be this way or is it a bug?

Cheers

**Comment by telbizov** [comment]
Hey @folke, would you say this is expected or a bug?

**Reply by folke** [reply]
Might be because some types are filtered. You can check with `:Trouble lsp_document_symbols` instead of `:Trouble symbols`.
The first one doesn't apply any filters.

**Reply by telbizov** [reply]
`lsp_document_symbols` worked great! Thank you
